claim-check/code-samples/sample-2/client-consumer/EventReader.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Newtonsoft.Json.Linq;
+
+namespace ClientConsumer
+{
+    class EventReader : IReader
+    {
+        private string downloadDestination;
+       
+        private EventProcessorClient processor;
+
+        public void Configure()
+        {
+            Console.WriteLine("Validating settings...");
+            foreach (string option in new string[] { "EventHubConnectionString", "StorageConnectionString", "BlobContainerName", "DownloadDestination" })
+            {
+                if (string.IsNullOrEmpty(ConfigurationManager.AppSettings?[option]))
+                {
+                    Console.WriteLine($"Missing '{option}' in App.config.");
+                    return;
+                }
+            }
+
+            string storageConnectionString = ConfigurationManager.AppSettings["StorageConnectionString"];
+            string eventhubConnectionString = ConfigurationManager.AppSettings["EventHubConnectionString"];
+            downloadDestination = ConfigurationManager.AppSettings["DownloadDestination"];
+            string blobContainerName = ConfigurationManager.AppSettings["BlobContainerName"];
+            Console.WriteLine("Connecting to Storage account...");
+            BlobContainerClient blobContainerClient = new BlobContainerClient(storageConnectionString, blobContainerName);
+            Console.WriteLine("Connecting to EventHub...");
+            processor = new EventProcessorClient(blobContainerClient, EventHubConsumerClient.DefaultConsumerGroupName, eventhubConnectionString);
+        }
+
+        //As an example, we'll just log the exception to the console in try/catch block.
+        //For real-world scenarios, you should take action appropriate to your application.
+        public async Task ProcessMessages(CancellationToken cancellationToken)
+        {
+            Console.WriteLine("The application will now start to listen for incoming messages.");
+
+            Task partitionInitializingHandler(PartitionInitializingEventArgs eventArgs)
+            {
+                if (eventArgs.CancellationToken.IsCancellationRequested)
+                {
+                    return Task.CompletedTask;
+                }
+                try
+                {
+                    var utcNow = DateTime.UtcNow;
```

Comment:
Not sure that I understand the benefit of reading this into a temporary variable, since you seem to be using it only once on line 61.    I'd suggest just accessing `DateTimeOffset.UtcNow` directly when creating the `EventPosition`.